cmake/FindFFT.cmake

```diff
@@ -29,7 +29,11 @@ endif()
 set(MKL_LINK static)
 # FIXME: Should we be setting the MPI implementation here so it can run with
 # things other than intelmpi?
-find_package(MKL CONFIG QUIET)
+
+if(NOT NESO_DISABLE_MKL)
+  find_package(MKL CONFIG QUIET)
```

Comment:
Personally I think MKL should be exception not the default (especially given how we call it once in the 1D code). So I think if the variables are a hard instruction to use MKL or not then the variable should be NESO_USE_MKL and if that's not set default to FFTW.

This approach keeps what we have as much as possible, i.e. try to find MKL, if not default to FFTW unless the user explicitly does not want MKL (because for whatever reason it doesn't work properly on their machine and they have no interest in the 1D code).